datafusion/optimizer/src/decorrelate_predicate_subquery.rs

```diff
@@ -54,7 +59,7 @@ impl DecorrelateWhereIn {
         predicate: &Expr,
         config: &dyn OptimizerConfig,
     ) -> Result<(Vec<SubqueryInfo>, Vec<Expr>)> {
-        let filters = split_conjunction(predicate); // TODO: disjunctions
+        let filters = split_conjunction(predicate); // TODO: add ExistenceJoin to support disjunctions
```

Comment:
This PR already support disjunction of scalar/in subquery?

Reply:
No, it doesn't.  To support disjunction of predicate subqueries, need to add a specific join type  `ExistenceJoin`

Reply:
Ok, BTW, there is a issue to track it `MarkJoin` #5368